AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -5468,6 +5470,313 @@ function New-AzDnsForwarder {
+                if($newshare -eq $null)
+                {
+                    Write-Verbose -Message  "The Destination Share doesn't exist. Creating a new share with the name provided by the user"
+                    New-AzStorageShare -Name $DestinationShareName -Context $storageAccountContext
+                }
+            }
+            #If SA is not AD connect, the user has to connect the SA to AD using the steps in the URL mentioned.
+            else
+            {
+                Write-Error -Message "To Proceed, you need to have Storage Account connected to Active Directory.
+                                        Refer to - https://docs.microsoft.com/en-us/azure/storage/files/storage-files-identity-auth-active-directory-enable"
+                            -Erroraction Stop
+            }
+        }
+        else
+        {
+            Write-Error -Message "The Storage Account doesn't exist. To create the Storage account and connect it to active directory, 
+                                    please follow the link https://docs.microsoft.com/en-us/azure/storage/files/storage-files-identity-auth-active-directory-enable" 
+                        -ErrorAction Stop
+        }
+
+        return $storageAccount
+    }
+}
+
+function ACLMigration 
+{
+    <#
+    .SYNOPSIS
+    Maps local share level permissions to Azure RBAC's role assignments.
+    .DESCRIPTION
+    Local share level permissions applied on domain users will be mapped to Azure RBAC role assignments applied on cloud identity corresponding to domain users.
+    .OUTPUTS
+    Boolean, True, if ACL migration can happen without any errors if CommitChanges is set to False.
+    .EXAMPLE
+    PS C:\> ACLMigration -LocalSharename "<localsharename>" - Destinationshare "<destinationshharename>" -ResourceGroupName "<resourceGroupName>" -StorageAccountName "<storageAccountName>" -CommitChanges $False -StopOnAADUserLookupFailure $True -AutoFitSharePermissionsOnAAD $True
+    #>
+
+    Param(
+         [Parameter(Mandatory=$true, Position=0, HelpMessage="Name of the share present on-prem.")]
+         [string]$LocalShareName,
+
+         [Parameter(Mandatory=$true, Position=1, HelpMessage="Name of the share on Azure storage account.")]
+         [string]$DestinationShareName,
+
+         [Parameter(Mandatory=$true, Position=2, HelpMessage="Resource group name of storage account.")]
+         [string]$ResourceGroupName,
+
+         [Parameter(Mandatory=$true, Position=3, HelpMessage="Storage account name on Azure")]
+         [string]$StorageAccountName,
+
+         [Parameter(Mandatory=$true, Position=4, HelpMessage="If false, the tool just checks for possible errors and reports back without making any changes on the cloud.")]
+         [bool]$CommitChanges,
+
+         [Parameter(Mandatory=$false, Position=5, HelpMessage="If true, ACL migration will be stopped upon failure to identity local user on AAD")]

```

Comment:
"to identity local user on AAD": identify?

Reply:
done